packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,244 @@
+                    supportsOrientationLockAsync:(NSString *)orientationLockString
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  EXOrientationLock orientationLock = [EXScreenOrientationUtilities stringToOrientationLock:orientationLockString];
+  UIInterfaceOrientationMask orientationMask = [EXScreenOrientationUtilities orientationLockJSToNative:orientationLock];
+  if (orientationMask == INVALID_MASK) {
+    resolve(@NO);
+  } else if ([EXScreenOrientationUtilities doesSupportOrientationMask:orientationMask]) {
+    resolve(@YES);
+  } else {
+    resolve(@NO);
+  }
+}
+
+UM_EXPORT_METHOD_AS(getOrientationAsync,
+                    getOrientationAsyncResolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve([EXScreenOrientationUtilities UIInterfaceOrientationToEXOrientation:_currentScreenOrientation]);
+}
+
+- (UIInterfaceOrientationMask)orientationMask
+{
+  return [[EXScreenOrientationModule sharedRegistry] orientationMaskForAppId:defaultAppId];
+}
+
+- (void)setOrientationMask:(UIInterfaceOrientationMask)mask
+{
+  return [[EXScreenOrientationModule sharedRegistry] setOrientationMask:mask forAppId:defaultAppId];
+}
+
++ (EXScreenOrientationRegistry *)sharedRegistry
+{
+  return (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
+}
+
+
+
+// Will be called when this module's first listener is added.
+- (void)startObserving
+{
+  _hasListeners = YES;
+}
+
+// Will be called when this module's last listener is removed, or on dealloc.
+- (void)stopObserving
+{
+//  hasListeners = NO;
+//  if ([[UIDevice currentDevice] isGeneratingDeviceOrientationNotifications]){
+//    [[UIDevice currentDevice] endGeneratingDeviceOrientationNotifications];
+//  }
+}
+
+- (void)handleDeviceOrientationChange:(NSNotification *)notification
+{
+  UIDevice *device = notification.object;
+  UIInterfaceOrientation currentDeviceOrientation = [EXScreenOrientationUtilities UIDeviceOrientationToUIInterfaceOrientation:[device orientation]];
+  UIInterfaceOrientationMask orientationMask = [self orientationMask];
+  
+  // first check: phone only rotates if device orientation is in mask, second check: we should send event only if device didn't rotate to current screen orientation
+  if ([EXScreenOrientationUtilities doesOrientationMask:orientationMask containOrientation:currentDeviceOrientation] && _currentScreenOrientation != currentDeviceOrientation) {
+    _currentScreenOrientation = currentDeviceOrientation;
+    if (_hasListeners) {
+      [self handleScreenOrientationChange];
+    }
+  }
+}
+
+- (void)handleScreenOrientationChange
+{
+  EXOrientationLock orientationLock = [EXScreenOrientationUtilities orientationLockNativeToJS:[self orientationMask]];
+  [_eventEmitter sendEventWithName:@"expoDidUpdateDimensions" body:@{
```

Comment:
Change event name to something without `expo` and `dimensions` 😅

Reply:
It's been called an `expoDidUpdateDimenstions` to be similar to RN event (`didUpdateDimensions`). So do we still want to rename it?